addons/beehave/nodes/beehave_root.gd

```diff
@@ -32,19 +32,23 @@ func _ready():
 	if actor_node_path:
 		actor = get_node(actor_node_path)
 
-	set_process_mode(self.process_mode)
 
 func _process(delta):
-	tick(delta)
-  
+	if process_mode == ProcessMode.IDLE:
```

Comment:
Wouldn't Godot just not run the _process function anyways if the process_mode is not IDLE? (same for PHYSICS_PROCESS)

Reply:
> void set_process ( [bool](https://docs.godotengine.org/en/3.5/classes/class_bool.html#class-bool) enable )
> 
> Enables or disables processing. When a node is being processed, it will receive a [NOTIFICATION_PROCESS](https://docs.godotengine.org/en/3.5/classes/class_node.html#class-node-constant-notification-process) on every drawn frame (and the [_process](https://docs.godotengine.org/en/3.5/classes/class_node.html#class-node-method-process) callback will be called if exists). Enabled automatically if [_process](https://docs.godotengine.org/en/3.5/classes/class_node.html#class-node-method-process) is overridden. Any calls to this before [_ready](https://docs.godotengine.org/en/3.5/classes/class_node.html#class-node-method-ready) will be ignored.
> 

**How & Why**
At first. When we override _process/_physic_process , They enabled automatically.
So both _process and _physic_process is call by Godot every tick.
In _process function ,if process_mode not IDLE, _process still call, but it skip tick, because the condition check not pass.(Same to _physic_process)
Set process_mode to MANUAL or anyother int value(not IDLE or PHYSICS_PROCESS), it skip tick in both _process and _physic_process. Then you have to call tick by manual.

set_process/set_physic_process still can effect behaveroot's  _process/_physic_process. Same as the Pause Mode can effect it too.

**Performance** 
Well, switch mode or disable process by set_process may have good performance. Because it make Godot not to call beehaveroot's _process in the engine side. By flag, Godot still call _process every time.But behaveroot's _process or _physic_process is just simple condition check, not effect too much. And flag base is more easier implement.